src/core/symbology-ng/qgsgraduatedsymbolrendererv2.cpp

```diff
@@ -1386,9 +1399,14 @@ void QgsGraduatedSymbolRendererV2::updateSymbols( QgsSymbolV2 *sym )
   int i = 0;
   foreach ( QgsRendererRangeV2 range, mRanges )
   {
-    QgsSymbolV2 *symbol = sym->clone();
-    symbol->setColor( range.symbol()->color() );
-    updateRangeSymbol( i, symbol );
+    QScopedPointer<QgsSymbolV2> symbol( sym->clone() );
```

Comment:
unless I'm missing something, this doesn't need to be a scoped pointer? (same with line 1352).